metric/common.go

```diff
@@ -54,9 +54,26 @@ type baseEntry interface {
 	read(t time.Time) metricdata.Point
 }
 
+func (bm *baseMetric) startTime() time.Time {
+	switch bm.bmType {
+	case cumulativeInt64:
+		return bm.start
+	case cumulativeFloat64:
+		return bm.start
+	case derivedCumulativeInt64:
+		return bm.start
+	case derivedCumulativeFloat64:
+		return bm.start
+	default:
+		// gauges don't have start time.
+		return time.Now()
```

Comment:
Better to return nil here, and check for nil in `read()`.

Reply:
done.